src/dune/lib.ml

```diff
@@ -498,32 +523,26 @@ module Lib_and_module = struct
     type nonrec t = t list
 
     let link_flags ts ~(lib_config : Lib_config.t) ~mode =
-      let libs =
-        List.filter_map ts ~f:(function
-          | Lib lib -> Some lib
-          | Module _ -> None)
-      in
       Command.Args.S
-        ( L.c_include_flags libs
```

Comment:
This change introduces a but that is later fixed in https://github.com/ocaml/dune/pull/2994

Reply:
Copying what I learned from a discussion in #2994:

Using `L.c_include_flags` for linking was a bit of a hack but it did the job for libraries that do not document their build artefacts properly.

Reply:
FWIW, we never intended to remove this. We thought `Command.Args.[ A "-I"; Path dir ]` in `link_flags` was sufficient to replace it, but we didn't think of this edge case. (I suspect we might also get the dependencies wrong in this case)